Readme: Add section about how translations work

tordans opened the pull request:
Closes https://github.com/openstreetmap/id-tagging-schema/issues/55

I had to search around for this – again – just now for https://github.com/bryceco/GoMap/issues/562. I think this should be clarified right in the Readme.

1ec5 commented on README.md:
```diff
@@ -17,6 +17,11 @@ This Tagging Schema fills that need, but with a number of caveats:
 
+## Translations
+
+* [Translations are managed in the Transiflex Project of the iD Editor](https://www.transifex.com/openstreetmap/id-editor/translate/#en_GB/presets/84278008).
```
> I don't see a problem with the signup page – it clearly states what you need to do and will lead you to a helpful page right after you signed up …

I’m logged into Transifex and am a translator in the Spanish and Vietnamese translation teams for iD. I’m not a member of the British English translation team, so all I get is a “Please select a language to get started”:

<img width="1440" alt="en-GB" src="https://user-images.githubusercontent.com/1231218/140011627-786e1773-dde3-42ca-8bee-5278ce060316.png">

If I were logged into Transifex but not a member of the OpenStreetMap organization at all, I would see an HTTP 404 error.

tordans replied:
Thanks, Minh, I tried to clarify at https://github.com/openstreetmap/id-tagging-schema/pull/245/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 without loosing the UX of actually finding where one needs to go.

tordans commented:
Looking into this with the Feedback of Minh in https://github.com/openstreetmap/id-tagging-schema/pull/245#discussion_r739155984, there are two more thinks I think we should look at

- [ ] The docs read as if anyone can translate. It that true? Just double checking if this is correct.
- [ ] How does the review process work? There was a GitHub issue a few weeks ago where someone requested access (cannot find it anymore). Should we explain this here in it's own section?
- [ ] As a follow up, we should also update https://github.com/openstreetmap/iD/blob/develop/CONTRIBUTING.md#translating with sub headlines and link to this new section for the part about presets.
- [ ] Are other information from https://github.com/openstreetmap/iD/blob/develop/CONTRIBUTING.md#translating relevant and should be added here? Like the `{variable}`-part?

1ec5 commented:
> The docs read as if anyone can translate. It that true? Just double checking if this is correct.

Transifex has a button to ask to join the translation team for a given locale. In the past, the administrators of this Transifex project would approve requests routinely, only rejecting requests for overly specific locales.

> How does the review process work?

Do you mean the review process for translations? Transifex does have a separate Reviewer role in order to mark translations as reviewed. This Transifex feature is generally only used for projects that rely on freelancers or contractors to provide initial translations. However, for Vietnamese, iD only pulls in reviewed translations: several years ago, someone used the Vietnamese locale as a dumping ground for their machine translation project, so this extra level of review prevented the machine translation from going live automatically.

> Are other information from https://github.com/openstreetmap/iD/blob/develop/CONTRIBUTING.md#translating relevant and should be added here? Like the `{variable}`-part?

Nothing in interim/source_strings.yaml currently uses this placeholder syntax. There is the `<…>` syntax used for `terms` strings; Transifex renders each of those placeholder as a token with a tooltip when you hover over it.

mbrzakovic commented:
Just seeing this is a duplicate of: #200. I think the additional info added here will be useful as well. When you find the time, please do make appropriate change.

tordans commented:
@1ec5 thanks for those answers. I updated the PR. I left out a some of the parts you answered since – now that I know – I think they are not too relevant to be explained to everyone.

I added a bullet to answer the question "when will I see my translations in iD(…)".

Another review would be great.

@mbrzakovic Would it be possible to squash-merge this and remove the part from #200 as part of resolving conflicts (in the github web UI)? If not, I have to checkout and rebase once I find time …

peternewman commented on README.md:
```diff
@@ -17,6 +17,11 @@ This Tagging Schema fills that need, but with a number of caveats:
+## Translations
+
+* [Translations are managed in the Transiflex Project of the iD Editor](https://www.transifex.com/openstreetmap/id-editor/translate/#en_GB/presets/84278008).
+* English translations for the `terms`-key should be added to the JSON data ([Exampple](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/natural/shrub.json#L16-L19)).
```
Although IIRC you'll only see a preview of the code with a hash and not with a branch name (I'm not sure about a tag).

1ec5 replied:
If you’re referring to an inline preview like:

https://github.com/openstreetmap/id-tagging-schema/blob/42d86acc7fa8e8ff01485437889c6784f244dea1/data/presets/natural/shrub.json#L16-L19

I think GitHub only does that in comments and the like, not in Markdown files.

peternewman commented:
> @mbrzakovic Would it be possible to squash-merge this and remove the part from #200 as part of resolving conflicts (in the github web UI)? If not, I have to checkout and rebase once I find time …

It's possible for the person doing the merge in GitHub to do the squash as part of the merge.

Or alternatively you can do it as part of a PR in your branch to do it all on the website.
